### PR TITLE
[csharp-netcore] Better deserialization

### PR DESCRIPTION
**Pull request by devhl-labs**
Fixes https://github.com/OpenAPITools/openapi-generator/issues/14916

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by devhl-labs** [comment]
Samples up-to-date check failed for unrelated samples/server/petstore/jaxrs-spec-required-and-readonly-property

**Comment by wing328** [comment]
that's fixed in the master so don't worry about it.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 1]
```diff
@@ -560,6 +560,11 @@ private void patchProperty(CodegenModel model, CodegenProperty property){
             property.datatypeWithEnum = property.datatypeWithEnum.replace("Dictionary>", property.items.datatypeWithEnum + ">");
             property.dataType = property.datatypeWithEnum;
         }
+
+        // HOTFIX: https://github.com/OpenAPITools/openapi-generator/issues/14944
+        if (property.datatypeWithEnum.equals("decimal")) {
+            property.isDecimal = true;
+        }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:567]: looks like something maybe wrong in the default codegen. i'll take a look later and add a test if needed.

**Comment by kYann** [comment]
This PR created a build issue, see https://github.com/OpenAPITools/openapi-generator/issues/14964

I think you should not use TryGet as it will create a silent error. Better to have the client throw an exception so we can check what's going on.

**Pull request merged**